src/plugins/analysis/cve_lookup/internal/setup_repository.py

```
@@ -0,0 +1,254 @@
+    dp.download_cpe(download_path=path)
+    if not glob(path + '*.xml'):
+        raise Exception('Error: Glob has found none of the specified files!')
+    return dp.extract_cpe(glob(path + '*.xml')[0])
+
+
+def init_cve_feeds_table(cve_list: List[CveEntry], table_name: str):
+    create(query='create_cve_table', table_name=table_name)
+    insert_into(query='insert_cve', table_name=table_name, input_data=dp.setup_cve_feeds_table(cve_list=cve_list))
+
+
+def init_cve_summaries_table(summary_list: list, table_name: str):
+    create(query='create_summary_table', table_name=table_name)
+    insert_into(query='insert_summary', table_name=table_name, input_data=dp.setup_cve_summary_table(summary_list=summary_list))
+
+
+def get_cve_import_content(cve_extraction_path: str, year_selection: list) -> Tuple[list, list]:
+    cve_list, summary_list = list(), list()
+    dp.download_cve(update=False, download_path=cve_extraction_path, years=year_selection)
+    for file in get_cve_json_files(cve_extraction_path):
+        cve_data, summary_data = dp.extract_cve(file)
+        cve_list.extend(cve_data)
+        summary_list.extend(summary_data)
+
+    return cve_list, summary_list
+
+
+def get_cve_update_content(cve_extraction_path: str) -> Tuple[list, list]:
+    dp.download_cve(update=True, download_path=cve_extraction_path, years=list())
+    cve_json_files = get_cve_json_files(cve_extraction_path)
+    if not cve_json_files:
+        raise Exception('Error: Glob has found none of the specified files!')
+    return dp.extract_cve(cve_json_files[0])
+
+
+def get_cve_json_files(cve_extraction_path: str) -> List[str]:
+    return glob(cve_extraction_path + 'nvdcve*.json')
+
+
+def cve_summaries_can_be_imported(extracted_summaries: list) -> bool:
+    return bool(extracted_summaries)
+
+
+def update_cve_repository(cve_extract_path: str):
+    if not exists(table_name='cve_table'):
+        print('\nCVE tables do not exist! Did you mean import CVE?\n')
```

:man_dancing: Don't you print on me

---

Also at a couple other spots in the code

---

If you're inclined to object: The prints used here (largely) refer to errors / exceptions. At some points like in `get_cpe_content` there is a nice raise of an exception instead of a print though. So keeping this behavior consistent would drop you down to a single `print` or `sys.stderr.write` in the **main** function.

---

I wouldn't dare to object. Replaced with exceptions and logging.